tests/test_examples.py

```diff
@@ -68,6 +71,62 @@ def test_download_examples() -> None:
     assert any(downloaded_examples_path.iterdir()), "Downloaded examples path is empty."
 
 
+@pytest.fixture
+def mock_s3_client(mocker):
+    mock = mocker.patch("boto3.client")
+    mock_instance = mock.return_value
+
+    def prepare_tar_buffer(contents):
+        tar_buffer = BytesIO()
+        with tarfile.open(fileobj=tar_buffer, mode="w:gz") as tar:
+            for content in contents:
+                tarinfo = tarfile.TarInfo(name=content["name"])
+                tarinfo.size = len(content["data"])
+                tar.addfile(tarinfo, BytesIO(content["data"]))
+        tar_buffer.seek(0)
+        return tar_buffer
+
+    return mock_instance, prepare_tar_buffer
+
+
+@pytest.mark.parametrize(
+    "contents, should_write",
+    [
+        ([{"name": "index.har", "data": b"dummy har content"}], True),
+        ([{"name": "largefile.bin", "data": b"x" * 10**6}], True),
+        ([], False),
+    ],
+)
+def test_download_har(mocker, mock_s3_client, contents, should_write):
```

Comment:
dont parameterize the should write, split it up here. it is an antipattern if you need if else assertions in your tests

Reply:
lets also add a with pytest.raises(....) test for an exception case